docs/source/training_tutorials/sft_lora_finetune_llm.mdx

```diff
@@ -0,0 +1,424 @@
+    ],
+    bias="none",
+    task_type="CAUSAL_LM",
+)
+
+# training_args is an instance of NeuronTrainingArguments
+args = training_args.to_dict()
+sft_config = NeuronSFTConfig(
+    max_seq_length=1024,
+    packing=False,
+    **args,
+)
+
+trainer = NeuronSFTTrainer(
+    args=sft_config,
+    model=model,
+    peft_config=config,
+    tokenizer=tokenizer,
+    train_dataset=dataset,
+    formatting_func=format_dolly,
+)
+
+# Start training
+trainer.train()
+
+trainer.save_model()  # Saves the tokenizer too for easy upload
+```
+
+The key points here are:
+
+- We use the `lazy_load_for_parallelism` context manager to lazily load the model. This will not load the full model weights on each worker, but instead only load the required weights (sharded or full). **This is much more memory efficient, and often mandatory to use.**
+- We define a `LoraConfig` that specifies which layers should have adapters, and the hyperparameters for theses adapters.
+- We create a [`~optimum.neuron.NeuronSFTConfig`] from regular `NeuronTrainingArguments`. Here we specify that we do not want to pack our examples, and that the max sequence length should be `1024`, meaning that every example will be either padded or truncated to a length of `1024`.
+- We use the [`~optimum.neuron.NeuronSFTTrainer`] to perform training. It will take the lazily loaded model, along with `lora_config`, `sft_config` and `format_dolly` and prepare the dataset and model for supervised fine-tuning.
+
+## 4. Launch Training 
+
+We prepared a script called [sft_lora_finetune_llm.py](https://github.com/huggingface/optimum-neuron/blob/main/docs/source/training_tutorials/lora_finetune_llm.py) summing up everything mentioned in this tutorial.
+
+PyTorch Neuron uses `torch_xla`. It evaluates operations lazily during execution of the training loops, which means it builds a symbolic graph in the background and the graph is executed on the hardware only when the tensor is printed, transfered to CPU, or `xm.mark_step()` is called. During execution, multiple graphs can be built depending on control-flow and it can take time to compile each graph sequentially. To alleviate that, the Neuron SDK provides `neuron_parallel_compile`, a tool which performs a fast trial run that builds all the graphs and compile them in parallel. This step is usually called precompilation.
+
+### Precompilation
+
+When training models on AWS Trainium we first need to compile our model with our training arguments. 
+
+To overcome this, we added a [model cache repository](https://huggingface.co/docs/optimum-neuron/guides/cache_system), which allows us to use precompiled models from the Hugging Face Hub to skip the compilation step. But be careful: every change in the model configuration might lead to a new compilation, which could result in some cache misses.
+
+_Note: If your model configuration is not cached please open an issue on [Github](https://github.com/huggingface/optimum-neuron/issues), we are happy to include it._
```

Comment:
We shall guide to the cache repo on the hub or not?

Reply:
I added a link!